src/System.Runtime.Extensions/tests/System/IO/PathTests_Unix.cs

```diff
@@ -28,22 +28,31 @@ public void GetFileName_Volume(string path, string expected)
             Assert.Equal(expected, new string(Path.GetFileName(path.AsSpan())));
         }
 
+        public static TheoryData<string, string> GetTempPath_SetEnvVar_Data => new TheoryData<string, string>
+        {
+            { "/tmp/", "/tmp" },
+            { "/tmp/", "/tmp/" },
+            { "/", "/" },
+            { "/var/tmp/", "/var/tmp" },
+            { "/var/tmp/", "/var/tmp/" },
+            { "~/", "~" },
+            { "~/", "~/" },
+            { ".tmp/", ".tmp" },
+            { "./tmp/", "./tmp" },
+            { "/home/someuser/sometempdir/", "/home/someuser/sometempdir/" },
+            { "/home/someuser/some tempdir/", "/home/someuser/some tempdir/" },
+            { "/tmp/", null },
+        };
+
         [Theory,
-            InlineData("/tmp/", "/tmp"),
-            InlineData("/tmp/", "/tmp/"),
-            InlineData("/", "/"),
-            InlineData("/var/tmp/", "/var/tmp"),
-            InlineData("/var/tmp/", "/var/tmp/"),
-            InlineData("~/", "~"),
-            InlineData("~/", "~/"),
-            InlineData(".tmp/", ".tmp"),
-            InlineData("./tmp/", "./tmp"),
-            InlineData("/home/someuser/sometempdir/", "/home/someuser/sometempdir/"),
-            InlineData("/home/someuser/some tempdir/", "/home/someuser/some tempdir/"),
-            InlineData("/tmp/", null)]
+            MemberData(nameof(GetTempPath_SetEnvVar_Data))]
         public void GetTempPath_SetEnvVar_Unix(string expected, string newTempPath)
         {
-            GetTempPath_SetEnvVar("TMPDIR", expected, newTempPath);
+            RemoteInvoke((expectedString, newTempPathString) =>
```

Comment:
You can't pass null this way because it becomes an argument to the child process. Your options are (1) encode it and decode it somehow eg as the string "NULL". (2) do what you were doing before by converting to a [Fact]

I think (2) is easiest, and it's also much faster to run because it only launches a single child process instead of 10+

Reply:
> and it's also much faster to run because it only launches a single child process instead of 10+

And harder to debug when one falls / see a pattern in the failures as one failing falls the rest of them.

I prefer (1), but up to you.

Reply:
@anipik either is fine to me

Reply:
Me, too.